content/tutorials/spatial_transform_crs_2/index.Rmd

```diff
@@ -0,0 +1,495 @@
+
+```{r}
+points_31370 <- st_transform(points, "EPSG:31370")
+```
+
+Notice the completely different coordinate ranges between both CRSs:
+
+```{r collapse=TRUE}
+st_geometry(points)
+```
+
+
+```{r collapse=TRUE}
+st_geometry(points_31370)
+```
+
+A specific area shown with `ggplot2` in `EPSG:31370`:
+
+```{r ggplot_water_points}
+ggplot() + 
+  geom_sf(data = watercourses, fill = "lightblue", colour = "lightblue") +
+  geom_sf(data = points_31370, fill = "red", shape = 21) +
+  lims(x = c(154e3, 155e3), y = c(218e3, 219e3)) +
+  coord_sf(datum = "EPSG:31370")
+```
+
+### Enforcing a specific pipeline
+
+We can request available transformation pipelines with `sf_proj_pipelines()`:
+
+```{r}
+pipelines <- sf_proj_pipelines("EPSG:28992", "EPSG:31370")
+```
+
+One can further limit this by specifying the area of interest (`aoi` argument), which we didn't do here.
+
+It is a dataframe with following structure:
+
+```{r}
+glimpse(pipelines)
+```
+
+Some metadata about the pipelines:
+
+```{r paged.print=FALSE}
+pipelines %>% 
+  as_tibble %>% 
+  select(definition, description, grid_count, accuracy, instantiable)
+```
+
+We can see that the first one has best accuracy.
+Several pipelines are based on one or two grids for a better accuracy.
+
+When printing the `pipelines` object, its first row is printed in a more readable manner:
+
+```{r}
+pipelines
+```
+
+Since the rows of pipelines are sorted by `instantiable` and `accuracy`, it makes sense to designate the printed pipeline as _Best instantiable operation_.
+
+So the same information appears as when manually selecting and then printing, except for the number of 'candidate operations found':
+
+```{r}
+pipelines[1,]
+```
+
+If you'd like to get even more metadata on the available pipelines, including the WKT definitions, you can also run PROJ's [`projinfo`](https://proj.org/apps/projinfo.html) command line program from within R if it's available in your system `PATH`.
+
+```{r eval=FALSE}
+system("projinfo -s EPSG:28992 -t EPSG:31370 --spatial-test intersects -o WKT2:2019")
+```
+
+The *default* operation returned by PROJ's `projinfo`, based on the union of the source and target CRS's bounding box, appears to be just one pipeline with so-called [ballpark accuracy](https://proj.org/glossary.html#term-Ballpark-transformation).
+It's obtained by dropping the `--spatial-test intersects` option (or replacing `intersects` by `contains`, which is default), and so it matches the pipeline where `accuracy` equals `NA` in our `pipelines` dataframe:
+
+```{r}
+pipelines[13,]
+```
+
+The low accuracy of a ballpark transformation is because datum differences between source and target CRSs are neglected:
+
+> It does not attempt any datum shift, hence the "ballpark" qualifier in its name.
+> Its accuracy is unknown, and could lead in some cases to errors of a few hundreds of metres.
+
+*(Source: <https://proj.org/glossary.html#term-Ballpark-transformation>)*
+
```

Comment:
Not sure about the added value of this section. If it has little practical use, I would delete it.

Reply:
I'd keep at least the part about the ballpark transformation. It's being applied by `st_transform()` in a part of the area for the `points` example, when not enforcing a single pipeline.

Reply:
Good idea!